Providers/Profile.swift

```diff
@@ -693,11 +729,23 @@ public class BrowserProfile: Profile {
             // We run this in the background after a few hundred milliseconds;
             // it doesn't really matter when it runs, so long as it doesn't
             // happen in the middle of a sync. We take the lock to prevent that.
+            let nothingSynced: [EngineStatus] = []
+
             self.doInBackgroundAfter(millis: 300) {
-                OSSpinLockLock(&self.syncLock)
-                self.handleRecreationOfDatabaseNamed(name).upon { res in
-                    log.debug("Reset of \(name) done: \(res.isSuccess)")
-                    OSSpinLockUnlock(&self.syncLock)
+                self.syncLock.lock()
+                defer { self.syncLock.unlock() }
+                // If a sync is not going already, make a dummy one.
+                let sync = self.beginSyncing() ? deferMaybe(nothingSynced) : self.currentSync!
```

Comment:
You've changed the behavior here.

We took the sync lock only to make sure that we didn't run during a sync. With these changes we'll notify begin/end of a sync. Can we simplify this?